gi-gtk-declarative-app-simple/src/GI/Gtk/Declarative/App/Simple.hs

```diff
@@ -81,24 +83,19 @@ run app = do
   assertRuntimeSupportsBoundThreads
   void $ Gtk.init Nothing
 
-  -- If any exception happen in `runLoop`, it will be re-throw here
+  -- If any exception happen in `runLoop`, it will be re-thrown here
   -- and the application will be killed.
-  -- NOTE: the use of Aync.async for `Gtk.main` is *MANDATORY*. The documentation for `
-  -- 'Async.concurrently' says that the first exception 'Async.cancel'
-  -- the second thread only at the condition that it is not a foreign
-  -- call, which is the case for 'Gtk.main'.
-  -- By using a second `Async.async` call, the wrapping thread can be killed.
-  fst <$> Async.concurrently (runLoop app <* Gtk.mainQuit) (Async.async Gtk.main)
+  main <- Async.async Gtk.main
+  runLoop app `finally` (Gtk.mainQuit >> Async.wait main)
```

Comment:
This change was necessary to fix a weird intermittent test suite failure. I suspect the issue was either that `Gtk.mainQuit` was not always called (when `runLoop app` threw an exception), or that somehow cancelling `Gtk.main` didn't always work properly.

Reply:
Damned. Indeed. Sorry about that ;(

Reply:
I missed it too :smile: